backend/routes/user.js

```diff
@@ -0,0 +1,33 @@
+const router = require('express').Router();
+const User = require('../models/user.model');
+
+router.route('/').get((req, res) => {
+  User.find()
+    .then((users) => res.json(users))
+    .catch((err) => res.status(400).json(`Error: ${err}`));
+});
+
+router.route('/add').post((req, res) => {
+  const { firstName, lastName, email } = req.body;
+
+  const newUser = new User({ firstName, lastName, email });
+
+  newUser
+    .save()
+    .then(() => res.json('User added!'))
+    .catch((err) => res.status(400).json(`Error: ${err}`));
+});
+
+module.exports = router;
```

Comment:
```suggestion
const express = require('express');
const app = express();
const User = require('../models/user.model');

app.get('/', function (req, res) {
   User.find()
    .then((users) => res.json(users))
    .catch((err) => res.status(400).json(`Error: ${err}`));
});
 
app.get('/add', function(res, req){
const { firstName, lastName, email } = req.body;

const newUser = new User({ firstName, lastName, email });

  newUser
    .save()
    .then(() => res.json('User added!'))
    .catch((err) => res.status(400).json(`Error: ${err}`));

});
  
module.exports = router;
```

Added this as a suggestion, if this works its more consistent with the routing part of express docs : [https://expressjs.com/en/guide/routing.html]

Reply:
or will this create problems with the middleware?

Reply:
New update pushed =)